### PR TITLE
refactor: Rename Context on protocol-contracts

### PR DESCRIPTION
**Pull request by andresaiello**
### Summary
- Rename context to zContext

### CODE REVIEW

**Review by brewmaster012 — COMMENTED** [review]
@andresaiello do you intend to remove the bindings *.go files? node repo relies on that by importing them

**Comment by andresaiello** [comment]
> @andresaiello do you intend to remove the bindings *.go files? node repo relies on that by importing them

I run the generate command but looks like just remove it. If you can generate it and push it would be great

File: contracts/zevm/ZRC20.sol [hunk 1]
```diff
@@ -38,6 +38,14 @@ contract ZRC20 is Context, IZRC20, IZRC20Metadata, ZRC20Errors {
     string private _symbol;
     uint8 private _decimals;
 
+    function _msgSender() internal view virtual returns (address) {
```
Comment (brewmaster012) [inline contracts/zevm/ZRC20.sol:41]: @andresaiello does this change ABI of the ZRC20 contract? 
I suppose we don't want to migrate ZRC20 contract on a3, so it's important not to change ABI of this contract.

Comment (brewmaster012) [reply contracts/zevm/ZRC20.sol:41]: otherwise it looks good to me.

Comment (andresaiello) [reply contracts/zevm/ZRC20.sol:41]: Changes the bytecode but not the public abi. I can revert it if you need the same bytecode but if we can avoid having two object with same name and different struct I think is better to change it

**Comment by brewmaster012** [comment]
Bytecode change is fine, as long as behavior and abi do not change.

________________________________

**Review by brewmaster012 — APPROVED** [review]

**Pull request merged**